avocado/utils/pci.py

```diff
@@ -0,0 +1,291 @@
+    :parm part: part of pci id.
+
+    :return: specific pci id of a pci device.
+    """
+    cmd = "lspci -Dnvmm -s %s | grep -iw %s" % (device, part)
+    output = process.system_output(cmd, shell=True, ignore_status=True)
+    if not output:
+        return
+    return output.split()[-1]
+
+
+def get_pci_id(device):
+    """
+    Gets pci id of given divice.
+
+    :parm device: pci device.
+
+    :return: pci id of a pci device.
+    """
+    pci_id = ""
+    for params in ['Vendor', 'Device', 'SVendor', 'SDevice']:
+        output = get_specific_pci_id(device, params)
+        if output:
+            pci_id += "%s:" % output
+    if pci_id:
+        return pci_id[:-1]
+
+
+def get_memory_address(device):
+    """
+    Gets the memory address of a pci device.
+
+    :parm device: pci device.
+
+    :return: memory address of a pci device.
+    """
+    cmd = "lspci -bv -s %s" % device
+    output = process.system_output(cmd, shell=True, ignore_status=True)
+    if not output:
+        return
+    for line in output.splitlines():
+        if 'Memory at' in line:
+            return "0x%s" % line.split()[2]
+
+
+def get_mask(device):
+    """
+    Gets the mask of pci device.
+
+    :parm device: pci device.
+
+    :return: mask of a pci device.
+    """
+    cmd = "lspci -vv -s %s" % device
+    output = process.system_output(cmd, shell=True, ignore_status=True)
+    if not output:
+        return
+    dic = {'K': '10', 'M': '20', 'G': '30'}
+    for line in output.splitlines():
+        if 'Region' in line and 'Memory at' in line:
+            val = line.split('=')[-1].split(']')[0]
+            memory_size = int(val[:-1]) * pow(2, int(dic[val[-1]]))
+            break
+    mask = hex((memory_size - 1) ^ int("0xffffffff", 16))
+    return mask
+
+
+def get_pci_devices():
+    """
+    Gets list of pci devices in the system.
+
+    :return: list of pci devices.
+    """
+    cmd = "lspci -D | grep -iv 'pci bridge' | awk '{print $1}'"
+    devices = process.system_output(cmd, shell=True, ignore_status=True)
+    if devices:
+        return devices.splitlines()
+
+
+def get_vpd(device):
+    """
+    Gets the vpd of the given device.
+
+    :parm device: pci device.
+
+    :return: dictionary of vpd of a pci device.
+    """
+    cmd = "lsvpd -l %s" % device
+    vpd = process.system_output(cmd, shell=True)
+    vpd_dic = {}
+    dev_list = []
+    for line in vpd.splitlines():
+        if len(line) < 5:
+            continue
+        if '*YL' in line:
+            vpd_dic['slot'] = line[4:]
+        elif '*DS' in line:
+            vpd_dic['pci_id'] = line[4:]
+        elif '*FC' in line:
+            vpd_dic['feature_code'] = line[4:]
+        elif '*AX' in line:
+            if not (device in line or vpd_dic['pci_id'].split()[0] in line):
+                dev_list.append(line[4:])
+    vpd_dic['devices'] = dev_list
+    return vpd_dic
+
+
+def get_cfg(device):
+    """
+    Gets the cfg data of the given device.
```

Comment:
I understand that the `get_vpd` probably relates to `hypervkvpd`, right? But I don't have any idea what `lscfg` is and where to get it. It'd be nice to include some details in the docstring.